zingolib/src/wallet/notes/interface.rs

```diff
@@ -42,35 +44,45 @@ pub trait OutputInterface: Sized {
     fn value(&self) -> u64;
 
     /// If the funds are spent, the TxId and Blockheight of record
-    fn spent(&self) -> &Option<(TxId, u32)>;
+    fn spend(&self) -> &Option<(TxId, ConfirmationStatus)>;
 
     /// Mutable access to the spent field.. hmm  NOTE:  Should we keep this pattern?
     /// what is spent becomes a Vec<OnceCell(TxiD, u32)>, where the last element of that
     /// Vec is the last known block chain record of the spend.  So then reorgs, just extend
     /// the Vec which tracks all BlockChain records of the value-transfer
-    fn spent_mut(&mut self) -> &mut Option<(TxId, u32)>;
+    fn spend_mut(&mut self) -> &mut Option<(TxId, ConfirmationStatus)>;
 
-    /// The TxId and broadcast height of a transfer that's not known to be on-record on the chain
-    fn pending_spent(&self) -> &Option<(TxId, u32)>;
+    /// returns the id of the spending transaction, whether pending or no
+    fn spending_txid(&self) -> Option<TxId> {
+        self.spend().map(|(txid, _status)| txid)
+    }
 
-    /// TODO: Add Doc Comment Here!
-    fn pending_spent_mut(&mut self) -> &mut Option<(TxId, u32)>;
+    /// Returns true if the note has ever been spent.
```

Comment:
But it actually returns a `SpendStatus`.

Reply:
fixed